Sources/ArcGISToolkit/Components/OfflineMapAreasView/OfflineMapAreasView.MapViewModel.swift

```diff
@@ -0,0 +1,168 @@
+        var canShowNotifications = false
+        
+        /// The job manager.
+        let jobManager = JobManager.shared
+        
+        init(map: Map) {
+            offlineMapTask = OfflineMapTask(onlineMap: map)
+            
+            if let portalItemID = map.item?.id?.rawValue {
+                FileManager.default.createDirectories(for: portalItemID)
+                
+                preplannedDirectory = FileManager.default.preplannedDirectory(portalItemID: portalItemID)
+            } else {
+                preplannedDirectory = FileManager.default.documentsDirectory
+            }
+        }
+        
+        /// Gets the preplanned map areas from the offline map task and creates the
+        /// offline map models.
+        func makePreplannedOfflineMapModels() async {
+            preplannedMapModels = await Result {
+                try await offlineMapTask.preplannedMapAreas
+                    .sorted(using: KeyPathComparator(\.portalItem.title))
+                    .compactMap {
+                        PreplannedMapModel(
+                            preplannedMapArea: $0,
+                            offlineMapTask: self.offlineMapTask,
+                            preplannedDirectory: self.preplannedDirectory
+                        )
+                    }
+            }
+            if let models = try? preplannedMapModels?.get() {
+                hasPreplannedMapAreas = !models.isEmpty
+            }
+        }
+        
+        /// Loads locally stored mobile map packages for the map's preplanned map areas.
+        func loadPreplannedMobileMapPackages() {
+            // Create mobile map packages with saved mmpk files.
+            let mmpkFiles = searchFiles(in: preplannedDirectory, with: "mmpk")
+            
+            self.mobileMapPackages = mmpkFiles.map(MobileMapPackage.init(fileURL:))
+            
+            // Pass mobile map packages to preplanned map models.
+            if let models = try? preplannedMapModels?.get() {
+                models.forEach { model in
+                    if let mobileMapPackage = mobileMapPackages.first(where: {
+                        $0.fileURL.deletingPathExtension().lastPathComponent == model.preplannedMapArea.id?.rawValue
+                    }) {
+                        model.setMobileMapPackage(mobileMapPackage)
+                    }
+                }
+            }
+        }
+        
+        /// Searches for files with a specified file extension in a given directory and its subdirectories.
+        /// - Parameters:
+        ///   - directory: The directory to search.
+        ///   - fileExtension: The file extension to search for.
+        /// - Returns: An array of file paths.
+        func searchFiles(in directory: URL, with fileExtension: String) -> [URL] {
+            guard let enumerator = FileManager.default.enumerator(
+                at: directory,
+                includingPropertiesForKeys: [.isDirectoryKey],
```

Comment:
you don't ever check the isDirectory resource key of the URL, so you don't need this
```suggestion
                includingPropertiesForKeys: nil,
```